### DISCUSSION TITLE
How to count tokens in Agency Swarm? (Token Accounting)

### DISCUSSION THREAD

**Discussion by hakimmanik** [VRSEN/agency-swarm#91] [Q&A] [answered]
Agency Swarm is great framework, but for my Saas, I'm charging my users depending on tokens. so I need to account how much token have I even used? I'd need both input tokens and output tokens used to generate one message reply (including messages sent to other agents)

**Comment by ivanguerron21** [comment]
Hi @hakimmanik, I was wondering if you were able to solve your issue and how you did it

**Comment by nayankr8625** [comment]
Hii @hakimmanik  Have youi encountered any solutions yet. If yes please help me with it

**Comment by ismail-cognios** [comment]
@hakimmanik Have you tried to use tiktoken python library?

**Comment by nicko-ai** [comment]
We’re currently integrating Langfuse observability, which will provide several options for tracking tokens and costs.

**Reply by nicko-ai** [reply, marked as answer]
We've released observability - see the announcement here https://www.skool.com/agency-ai/observability-is-heresay-hello-to-langfuse?p=73ad37d8
